### ISSUE TITLE
Adding additional context into views/urls causes problems

### ISSUE THREAD

**Issue by 9mido** [pennersr/django-allauth#2740]
Password Change Error:
AttributeError at /accounts/password/change/ 'AnonymousUser' object has no attribute 'has_usable_password'

I am logged in trying to change the user's password but I am getting an AnonymousUser error because I have ACCOUNT_LOGOUT_ON_PASSWORD_CHANGE = True. Changing to False solved the problem but that probably should not happen and may need to be re-written to account for when True. 

All I tried to do is add extra context of the captcha public key environment variable to the password change view so that the the key could get passed to the password change template. 

Also, I am having a similar problem with password resets where it redirects to /accounts/password/reset/done/ url but not the /accounts/password/reset/done/ template but the password reset email still gets sent. I am using the same code I have below for password resets but the form names are changed. 

Even something like this is giving me a problem:

```
from allauth.account.views import SignupView, LoginView, PasswordChangeView, PasswordResetView
url('^accounts/password/reset/', PasswordResetView.as_view(extra_context={'captcha_public_key':settings.CAPTCHA_PUBLIC_KEY}), name="account_reset_password"),
```

It may be happening in other places within allauth. 

Update: A context processor seems to do the trick. But it would have been nice to be able to do it the way outlined here in the issue.
https://stackoverflow.com/questions/433162/can-i-access-constants-in-settings-py-from-templates-in-django

Urls.py:

```
from app_name.views import CaptchaPasswordChangeView
url('^accounts/password/change/', CaptchaPasswordChangeView.as_view(), name="account_change_password"),

```

Views.py:

```
class CaptchaPasswordChangeView(PasswordChangeView):
    def get_context_data(self, **kwargs):
        captcha_public_key = settings.CAPTCHA_PUBLIC_KEY
        context = super(CaptchaPasswordChangeView, self).get_context_data(**kwargs)
        context["captcha_public_key"] = captcha_public_key
        return context
```

Html:

```
    <form id="password_change" method="POST" action="{% url 'account_change_password' %}" autocomplete="off" class="password_change">
        {% csrf_token %}
        {{ form.oldpassword|as_crispy_field }}
        {{ form.password1|as_crispy_field }}
        {{ form.password2|as_crispy_field }}

      <script src="https://www.google.com/recaptcha/api.js?render={{captcha_public_key}}"></script>
      <script>
        grecaptcha.ready(function() {
          var grecaptcha_execute = function(){
            grecaptcha.execute('{{captcha_public_key}}', {action: "{% url 'account_change_password' %}"}).then(function(token) {
              //document.getElementById('password_change').insertAdjacentHTML('beforebegin', "<input type='hidden' name='recaptcha_token'>");
              document.querySelectorAll('input.django-recaptcha-hidden-field').forEach(function (value) {
                  value.value = token;
              });
                  const previous = document.getElementById('password_change').querySelector('[name="recaptcha_token"]'); if (previous) { previous.remove(); }
                  document.getElementById('password_change').insertAdjacentHTML('afterbegin', '<input type="hidden" name="recaptcha_token" value="' + token + '">');
              return token;
            });
          };
          grecaptcha_execute();
          setInterval(grecaptcha_execute, 120000);
        });
      </script>
        <button id="password-change" type="submit" class="form-control" name="action">{% trans "Change Password" %}</button>
    </form>
```

Traceback:

```
Internal Server Error: /accounts/password/change/
Traceback (most recent call last):
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/core/handlers/base.py", line 179, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/views/decorators/debug.py", line 89, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/allauth/account/views.py", line 592, in dispatch
    return super(PasswordChangeView, self).dispatch(request, *args, **kwargs)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/allauth/account/views.py", line 91, in get
    request, *args, **kwargs
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/views/generic/edit.py", line 133, in get
    return self.render_to_response(self.get_context_data())
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/allauth/account/views.py", line 595, in render_to_response
    if not self.request.user.has_usable_password():
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/utils/functional.py", line 241, in inner
    return func(self._wrapped, *args)
  File "/home/user/.local/share/virtualenvs/project_name/lib/python3.7/site-packages/django/utils/functional.py", line 241, in inner
    return func(self._wrapped, *args)
AttributeError: 'AnonymousUser' object has no attribute 'has_usable_password'
```

**Comment by pennersr** [comment]
Fixed via 7d718553

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2740
